dali/kernels/tensor_shape.h

```diff
@@ -612,7 +640,7 @@ struct TensorListShape<DynamicDimensions>
   }
 
   int sample_dim() const { return dim; }
-  int size() const { return shapes.size() / sample_dim(); }
+
   void set_sample_dim(int dim) { this->dim = dim; }
```

Comment:
A side note:
How about making it private and befriending `TensorListShape` with its base class? This function leaves the object in an inconsistent state. I know that our data members our public, but that's part of our contract - sort of "you modify object directly, better know what you're doing" thing. Public functions, on the other hand, are expected to keep object usable.

Reply:
Done. Good idea.